lb_content_resolver/content_resolver.py

```diff
@@ -97,80 +103,78 @@ def traverse(self, relative_path):
 
         return True
 
-    def encode_string(self, text):
-        if text is None:
-            return None
-        return unidecode(re.sub(" +", " ", re.sub(r'[^\w ]+', '', text)).strip().lower())
+    def build_index(self):
+        """
+            Fetch the data from the DB and then build the fuzzy lookup index.
+        """
 
-    def resolve_recording(self, artist_name, recording_name, distance=2):
+        artist_recording_data = []
+        for recording in Recording.select():
+            artist_recording_data.append((recording.artist_name, recording.recording_name, recording.id))
 
-        query = "%s~%d %s~%d" % (self.encode_string(artist_name), distance, self.encode_string(recording_name), distance)
-        self.open_index()
+        self.fuzzy_index = FuzzyIndex(self.index_dir)
+        self.fuzzy_index.build(artist_recording_data)
 
-        ret = []
-        with self.ix.searcher() as searcher:
-            parser = QueryParser("lookup", self.ix.schema)
-            parser.add_plugin(FuzzyTermPlugin())
-            query = parser.parse(query)
-            for result in searcher.search(query):
-                ret.append(dict(result))
+    def encode_string(self, text):
+        """ 
+            Remove unwanted crap from the query string and only keep essential information.
 
-        return ret
+            'This is the ultimate track !!' -> 'thisistheultimatetrack'
+        """
+        if text is None:
+            return None
+        return unidecode(re.sub(" +", " ", re.sub(r'[^\w ]+', '', text)).strip().lower())
 
-    def resolve_playlist(self, jspf_playlist, m3u_playlist):
-        return convert_jspf_to_m3u(self, jspf_playlist, m3u_playlist)
+    def resolve_playlist(self, jspf_playlist, m3u_playlist, match_threshold):
+        """ 
+            Open the database, build the fuzzy index and then resolve the playlist.
+        """
+        self.open_db()
+        self.build_index()
+        return convert_jspf_to_m3u(self.fuzzy_index, jspf_playlist, m3u_playlist, match_threshold)
+
+    def add_or_update_recording(self, mdata):
+        """ 
+            Given a Recording, add it to the DB if it does not exist. If it does,
+            update the recording instead
+        """
 
-    def lookup_metadata(self, file_path):
+        with db.atomic() as transaction:
+            try:
+                recording = Recording.select().where(Recording.file_path == mdata['file_path']).get()
+            except:
+                recording = Recording.create(file_path=mdata['file_path'],
+                                             artist_name=mdata["artist_name"],
+                                             release_name=mdata["release_name"],
+                                             recording_name=mdata["recording_name"],
+                                             name=mdata["recording_name"],
```

Comment:
Why is there both `name` and `recording_name` with the same value?

Reply:
#copypasta